File: src/PortaleUser/OrderEntryPlanner/Components/PortalUser.cs
using DI.PortalUser2;
using DI.PortalUser2.Data;
using System;

namespace OrderEntryPlanner.Components
{

	/// <summary>
	/// Classe SINGLETON inizializzata nel global.asax per condividere le property in tutto il progetto.
	/// </summary>
	public sealed class PortalUserSingleton
	{

		#region PublicProperties

		public readonly RoleManagerUtility2 RoleManagerUtility;
		public readonly SessioneUtente SessioneUtente;
		public readonly PortalDataAdapterManager PortalDataAdapterManager;

		#endregion

		private PortalUserSingleton()
		{
			//Ottengo la connection string del database DiPortalUser.
			string portalUserConnectionString = Properties.Settings.Default.PortalUserConnectionString;

			//Ottengo la connection string del database SAC.
			string sacConnectionString = Properties.Settings.Default.SacConnectionString;

			//Ottengo l'utente SAC.
			string SacWs_User = Properties.Settings.Default.SacWs_User;

			//Ottengo la password SAC.
			string SacWs_Password = Properties.Settings.Default.SacWs_Password;

			//Controllo che "portalUserConnectionString" sia valorizzata.
			if (string.IsNullOrEmpty(portalUserConnectionString))
			{
				throw new Exception("Parametro di configurazione assente: PortalUserConnectionString.");
			}

			//Controllo che "sacConnectionString" sia valorizzata.
			if (string.IsNullOrEmpty(sacConnectionString))
			{
				throw new Exception("Parametro di configurazione assente: SacConnectionString.");
			}

			SessioneUtente = new SessioneUtente(sacConnectionString, portalUserConnectionString, SacWs_User, SacWs_Password);
			RoleManagerUtility = new RoleManagerUtility2(portalUserConnectionString, sacConnectionString, SacWs_User, SacWs_Password);
			PortalDataAdapterManager = new PortalDataAdapterManager(portalUserConnectionString);
		}

		private static PortalUserSingleton _instance;

		public static PortalUserSingleton instance
		{
			get
			{
				if (_instance == null)
				{
					_instance = new PortalUserSingleton();
				}
				return _instance;
			}
		}


	}
}

File: src/SAC/SAC.RaccoltaConsensi/ConsensiPaziente/Code/PortalUser.vb
Imports DI.PortalUser2
Imports DI.PortalUser2.Data
Imports System

'<summary>
'Classe SINGLETON inizializzata nel global.asax per condividere le property in tutto il progetto.
'</summary>
Public NotInheritable Class PortalUserSingleton

#Region "PublicProperties"
    Public ReadOnly RoleManagerUtility As RoleManagerUtility2
    Public ReadOnly SessioneUtente As SessioneUtente
    Public ReadOnly PortalDataAdapterManager As PortalDataAdapterManager
#End Region

    Private Sub New()
        'Ottengo la connection string del database DiPortalUser.
        Dim portalUserConnectionString As String = My.Settings.PortalUserConnectionString

        'Ottengo la connection string del database SAC.
        Dim sacConnectionString As String = System.Configuration.ConfigurationManager.ConnectionStrings("SACConnectionString").ConnectionString

        'Ottengo l'utente SAC.
        Dim SacWs_User As String = My.Settings.SACUsername

        '//Ottengo la password SAC.
        Dim SacWs_Password As String = My.Settings.SACPassword

        '	//Controllo che "portalUserConnectionString" sia valorizzata.
        If String.IsNullOrEmpty(portalUserConnectionString) Then
            Throw New Exception("Parametro di configurazione assente: PortalUserConnectionString.")
        End If

        'Controllo che "sacConnectionString" sia valorizzata.
        If String.IsNullOrEmpty(sacConnectionString) Then
            Throw New Exception("Parametro di configurazione assente: SacConnectionString.")
        End If

        SessioneUtente = New SessioneUtente(sacConnectionString, portalUserConnectionString, SacWs_User, SacWs_Password)
        RoleManagerUtility = New RoleManagerUtility2(portalUserConnectionString, sacConnectionString, SacWs_User, SacWs_Password)
        PortalDataAdapterManager = New PortalDataAdapterManager(portalUserConnectionString)
    End Sub

    Private Shared _instance As PortalUserSingleton
    Public Shared ReadOnly Property instance As PortalUserSingleton
        Get
            If (_instance Is Nothing) Then
                _instance = New PortalUserSingleton()
            End If
            Return _instance
        End Get
    End Property

End Class
